### PR TITLE
RAB: Integrate staging tests for the .map method

### PR DESCRIPTION
**Pull request by ioannad**
of Array.prototype and TypedArray.prototype

This is part of PR #3888 to make reviewing easier. Includes changes to use the helper ./harness/resizableArrayBufferUtils.js

### CODE REVIEW

File: test/built-ins/TypedArray/prototype/map/resizable-buffer-shrink-mid-iteration.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
+// Copyright 2023 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-%typedarray%.prototype.map
+description: >
+  TypedArray.p.map behaves correctly on TypedArrays backed by resizable buffers
+  that are grown mid-iteration.
```
Comment (ptomato) [inline test/built-ins/TypedArray/prototype/map/resizable-buffer-shrink-mid-iteration.js:8]:
```suggestion
  that are shrunk mid-iteration.
```

**Review by ptomato — APPROVED** [review]
:+1:, just two minor c&p errors in the descriptions.

**Comment by ioannad** [comment]
@ptomato  thanks!

**Review by ptomato — APPROVED** [review]
Thanks!

**Pull request merged**